core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -335,6 +337,33 @@ public ListStoresResponse listStores(ListStoresRequest.Filter filter) {
   @Transactional
   public ApplyFeatureSetResponse applyFeatureSet(FeatureSetProto.FeatureSet newFeatureSet)
       throws InvalidProtocolBufferException {
+    // Autofill default project if not specified
+    if (newFeatureSet.getSpec().getProject().isEmpty()) {
+      newFeatureSet =
+          newFeatureSet
+              .toBuilder()
+              .setSpec(newFeatureSet.getSpec().toBuilder().setProject(Project.DEFAULT_NAME).build())
+              .build();
+    }
+
+    String projectName = newFeatureSet.getSpec().getProject();
+    String featureSetName = newFeatureSet.getSpec().getName();
+    List<Boolean> isSubscribedToStores = new ArrayList<>() {};
+    for (Store store : storeRepository.findAll()) {
+      List<Subscription> subscriptionList = store.getSubscriptions();
+      boolean isSubscribed =
+          isSubscribedToFeatureSet(subscriptionList, projectName, featureSetName);
+      isSubscribedToStores.add(isSubscribed);
+    }
+    // Only throw error if FeatureSet is not subscribed by ALL stores
+    if (!isSubscribedToStores.isEmpty()
+        && isSubscribedToStores.stream().allMatch(x -> x == false)) {
+      throw new RegistrationException(
+          String.format(
+              "The supplied Project and FeatureSet, %s/%s has been blacklisted and is not available for registration.",
```

Comment:
It doesn't necessarily mean that it was blacklisted. There simply can be no subscribers.

Reply:
I think we need to give some hint here, like `ask your administrator to add subscription in store configuration on serving layer`

Reply:
Would this be clearer and more accurate then `has been blacklisted and is not...` -> `is either blacklist or not subscribed to and is not...`?